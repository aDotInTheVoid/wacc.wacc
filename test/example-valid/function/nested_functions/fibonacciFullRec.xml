<unit>
  <fn>
    <name>fibonacci</name>
    <args><arg>0</arg></args>
    <body>
      <if>
        <cond><local>0</local><num>1</num><op>le</op></cond>
        <when><local>0</local><return/></when>
        <else/>
      </if>
      <addr_local>1</addr_local>
      <local>0</local>
      <num>1</num>
      <op>sub</op>
      <call_func><name>fibonacci</name><nargs>1</nargs></call_func>
      <assign_do/>
      <addr_local>2</addr_local>
      <local>0</local>
      <num>2</num>
      <op>sub</op>
      <call_func><name>fibonacci</name><nargs>1</nargs></call_func>
      <assign_do/>
      <local>1</local>
      <local>2</local>
      <op>add</op>
      <return/>
    </body>
  </fn>
  <main>
    <str>"This program calculates the nth fibonacci number recursively."</str>
    <print><kind>str</kind><newline>true</newline></print>
    <str>"Please enter n (should not be too large): "</str>
    <print><kind>str</kind><newline>false</newline></print>
    <addr_local>0</addr_local>
    <num>0</num>
    <assign_do/>
    <addr_local>0</addr_local>
    <read><ty>int</ty></read>
    <str>"The input n is "</str>
    <print><kind>str</kind><newline>false</newline></print>
    <local>0</local>
    <print><kind>i32</kind><newline>true</newline></print>
    <str>"The nth fibonacci number is "</str>
    <print><kind>str</kind><newline>false</newline></print>
    <addr_local>1</addr_local>
    <local>0</local>
    <call_func><name>fibonacci</name><nargs>1</nargs></call_func>
    <assign_do/>
    <local>1</local>
    <print><kind>i32</kind><newline>true</newline></print>
  </main>
</unit>