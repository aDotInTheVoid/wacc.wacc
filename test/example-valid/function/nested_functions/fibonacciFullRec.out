This program calculates the nth fibonacci number recursively.
Please enter n (should not be too large): #input#
The input n is #output#
The nth fibonacci number is #output#
